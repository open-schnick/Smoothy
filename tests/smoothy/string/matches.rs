use crate::failing_assertion;
use regex::Regex;
use smoothy::{assert_that, AssertionConnector, StringAssertion};

#[test]
fn succeeds() {
    let _connector: AssertionConnector<&str> =
        assert_that("I categorically deny having triskaidekaphobia.")
            .matches(&Regex::new(r"\b\w{13}\b").unwrap());
}

#[allow(clippy::unnecessary_to_owned)]
#[test]
fn succeeds_with_string() {
    let _connector: AssertionConnector<String> =
        assert_that("I categorically deny having triskaidekaphobia.".to_string())
            .matches(&Regex::new(r"\b\w{13}\b").unwrap());
}

#[test]
fn fails() {
    failing_assertion!({
        assert_that("I deny having fun.".to_string()).matches(&Regex::new(r"\b\w{13}\b").unwrap());
    });
}
