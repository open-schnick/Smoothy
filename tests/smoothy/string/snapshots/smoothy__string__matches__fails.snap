---
source: tests/smoothy/string/matches.rs
expression: assertion_failed_output
---
assertion failed: `(Value is matching regex)`
           found:  "I deny having fun."
